src/go/configinfo/service_info.go

```diff
@@ -40,9 +39,12 @@ type ServiceInfo struct {
 
 	// An array to store all the api names
 	ApiNames []string
-	// A sorted array to store all the method name for this service.
-	// Should always iterate this array to avoid test fail due to order issue.
+
+	// A ordered slice of operation names. Follows the same order as the service config.
+	// Ordering is important, both for route matching and testability.
+	// All functions that output order-dependent configs should use this ordering.
```

Comment:
Could you also comment the first-win match of order dependency of route matching?

Now we are depending on the order from service config(OpenAPI compiler), which is likely to break. Can we we sort order by ourselves?

Reply:
1) I am not clear what order in service configs we are following?  apiMathod?  HttpRules?  Backend Rules? 
2) for allow=all wildcard,  are we sure they are in the last with such ordering?

Reply:
Done. See other comment about OpenAPI compiler. 

I highly recommend that we don't sort them, I feel that is even more brittle. And what if a user manually specifies their own http rule? Then our sorting could break them. It's best to take the order of the service config, which the user can easily modify if needed.

Reply:
SG